server2/Arcadia.Assistant/Arcadia.Assistant.Notifications/Notifications.cs

```diff
@@ -0,0 +1,149 @@
+namespace Arcadia.Assistant.Notifications
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Fabric;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Castle.Core.Internal;
+
+    using Contracts;
+    using Contracts.Models;
+
+    using DeviceRegistry.Contracts;
+    using DeviceRegistry.Contracts.Models;
+
+    using Employees.Contracts;
+
+    using Interfaces;
+
+    using Microsoft.Extensions.Logging;
+    using Microsoft.ServiceFabric.Services.Communication.Runtime;
+    using Microsoft.ServiceFabric.Services.Remoting.Runtime;
+    using Microsoft.ServiceFabric.Services.Runtime;
+
+    using PushNotifications.Contracts;
+    using PushNotifications.Contracts.Models;
+
+    using UserPreferences.Contracts;
+
+    /// <summary>
+    ///     An instance of this class is created for each service instance by the Service Fabric runtime.
+    /// </summary>
+    public class Notifications : StatelessService, INotifications
+    {
+        private readonly IDeviceRegistry deviceRegistry;
+        private readonly ILogger logger;
+
+        private readonly Dictionary<string, IEnumerable<NotificationType>> notificationProvidersMap;
+        private readonly INotificationSettings notificationSettings;
+        private readonly IPushNotifications pushNotifications;
+        private readonly IUsersPreferencesStorage userPreferences;
+
+        public Notifications(
+            StatelessServiceContext context,
+            INotificationSettings notificationSettings,
+            IDeviceRegistry deviceRegistry,
+            IUsersPreferencesStorage userPreferences,
+            IPushNotifications pushNotifications,
+            ILogger<Notifications> logger)
+            : base(context)
+        {
+            this.notificationSettings = notificationSettings;
+            this.deviceRegistry = deviceRegistry;
+            this.userPreferences = userPreferences;
+            this.pushNotifications = pushNotifications;
+            this.logger = logger;
+            this.notificationProvidersMap = this.notificationSettings.ClientNotificationProvidersMap;
+        }
+
+        public async Task Send(IEnumerable<EmployeeId> employeeId, NotificationMessage notificationMessage, CancellationToken cancellationToken)
+        {
+            if (employeeId.IsNullOrEmpty())
+            {
+                return;
+            }
+
+            var notificationProviders = this.notificationProvidersMap.TryGetValue(notificationMessage.ClientName, out var map) ? map : Enum.GetValues(typeof(NotificationType)).OfType<NotificationType>().ToList();
```

Comment:
why is it a 'client name' based mapping? Bad naming I'd say, there might be multiple notification types in one service. I'd rename it to NotificationType and leave it as is. For AppCenterBuilds I'd have its value equal 'New_Build_Available' or something like that. Probably it makes sense to make a short type alias like `class readonly NotificationType { public readonly Type {get; }  }` so it'd be much easier to search for different types in a future

Reply:
Changed to NotificationTemplate. Added new property for all notification types enumeration.